pycloudlib/ibm/instance.py

```diff
@@ -0,0 +1,735 @@
+        }
+        client.create_security_group_rule(
+            default_sg["id"], rule_proto
+        ).get_result()
+
+        subnet = _Subnet.create(
+            client,
+            name=f"{name}-subnet",
+            zone=zone,
+            resource_group_id=resource_group_id,
+            vpc_id=vpc["id"],
+        )
+        return cls(
+            *args,
+            client=client,
+            vpc=vpc,
+            resource_group_id=resource_group_id,
+            subnet=subnet,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_existing(
+        cls,
+        *args,
+        client: VpcV1,
+        name: str,
+        resource_group_id: str,
+        zone: str,
+        **kwargs,
+    ) -> "VPC":
+        """Find a VPC by name.
+
+        Try to discover a Subnet within it or create it if not found.
+        """
+        vpc = _get_first(
+            client.list_vpcs,
+            resource_name="vpcs",
+            filter_fn=lambda vpc: vpc["name"] == name,
+        )
+        if vpc is None:
+            raise IBMException(f"VPC not found: {name}")
+
+        try:
+            subnet = _Subnet.discover(client, vpc_id=vpc["id"])
+        except IBMException:
+            subnet = _Subnet.create(
+                client,
+                name=f"{name}-subnet",
+                zone=zone,
+                resource_group_id=resource_group_id,
+                vpc_id=vpc["id"],
+            )
+
+        return cls(
+            *args,
+            client=client,
+            vpc=vpc,
+            resource_group_id=resource_group_id,
+            subnet=subnet,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_default(
+        cls,
+        *args,
+        client: VpcV1,
+        resource_group_id: str,
+        region: str,
+        zone: str,
+        **kwargs,
+    ) -> "VPC":
+        """Find the `default` VPC and Subnet."""
+        default_name = f"{region}-default-vpc"
+        vpc = _get_first(
+            client.list_vpcs,
+            resource_name="vpcs",
+            filter_fn=lambda vpc: vpc["name"] == default_name,
+            resource_group_id=resource_group_id,
+        )
+        if vpc is None:
+            raise IBMException(f"VPC not found: {default_name}")
+        subnet = _Subnet.from_default(client, zone=zone, vpc_id=vpc["id"])
+        return cls(
+            *args,
+            client=client,
+            vpc=vpc,
+            subnet=subnet,
+            resource_group_id=resource_group_id,
+            **kwargs,
+        )
+
+    @property
+    def id(self) -> str:
+        """VPC ID."""
+        return self._vpc["id"]
+
+    @property
+    def name(self) -> str:
+        """VPC name."""
+        return self._vpc["name"]
+
+    @property
+    def subnet_id(self) -> str:
+        """Subnet ID."""
+        if self._subnet is None:
+            raise IBMException("No subnet available")
+        return self._subnet.id
+
+    def delete(self) -> None:
```

Comment:
This is only called in `examples/ibm.py`. VPC deletion is not called within pycloudlib.

I think if we support VPC creation, we have to support deletion, otherwise it would not be very useful for clients.

What do you suggest? Removing VPC creation and deletion, or keep it as is?

Reply:
I think it makes sense to keep it. I was more worried about the possibility of automatically deleting something (e.g., like we do with floating ip on instance deletion). As long as it's an explicit operation, I think it's ok.